MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -538,6 +577,373 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputHermiteSplineModel(vtkMRMLMarkups
+        treeStartIndex = v;
+        treeEndIndex = u;
+      }
+    }
+  }
+  // use the 1D vector as a 2D vector
+  std::vector<double*> graph(numPoints);
+  for (int v = 0; v < numPoints; v++)
+    graph[v] = &(distances[v * numPoints]);
+
+  // implementation of Prim's algorithm heavily based on:
+  // http://www.geeksforgeeks.org/greedy-algorithms-set-5-prims-minimum-spanning-tree-mst-2/
+  std::vector<int> parent(numPoints); // Array to store constructed MST
+  std::vector<double> key(numPoints);   // Key values used to pick minimum weight edge in cut
+  std::vector<bool> mstSet(numPoints);  // To represent set of vertices not yet included in MST
+ 
+  // Initialize all keys as INFINITE
+  for (int i = 0; i < numPoints; i++)
+  {
+    key[i] = VTK_DOUBLE_MAX;
+    mstSet[i] = false;
+  }
+ 
+  // Always include first 1st vertex in MST.
+  key[treeStartIndex] = 0.0;     // Make key 0 so that this vertex is picked as first vertex
+  parent[treeStartIndex] = -1; // First node is always root of MST 
+ 
+  // The MST will have numPoints vertices
+  for (int count = 0; count < numPoints-1; count++)
+  {
+    // Pick the minimum key vertex from the set of vertices
+    // not yet included in MST
+    int nextPointIndex = -1;
+    double minDistance = VTK_DOUBLE_MAX, min_index;
+    for (int v = 0; v < numPoints; v++)
+    {
+      if (mstSet[v] == false &&
+          key[v] < minDistance)
+      {
+        minDistance = key[v];
+        nextPointIndex = v;
+      }
+    }
+ 
+    // Add the picked vertex to the MST Set
+    mstSet[nextPointIndex] = true;
+ 
+    // Update key value and parent index of the adjacent vertices of
+    // the picked vertex. Consider only those vertices which are not yet
+    // included in MST
+    for (int v = 0; v < numPoints; v++)
+ 
+        // graph[u][v] is non zero only for adjacent vertices of m
+        // mstSet[v] is false for vertices not yet included in MST
+        // Update the key only if graph[u][v] is smaller than key[v]
+      if (graph[nextPointIndex][v] >= 0 && 
+          mstSet[v] == false &&
+          graph[nextPointIndex][v] <  key[v])
+      {
+        parent[v] = nextPointIndex;
+        key[v] = graph[nextPointIndex][v];
+      }
+  }
+ 
+  // determine the "trunk" path of the tree, from first index to last index
+  std::vector<int> pathIndices;
+  int currentPathIndex = treeEndIndex;
+  while (currentPathIndex != -1)
+  {
+    pathIndices.push_back(currentPathIndex);
+    currentPathIndex = parent[currentPathIndex]; // go up the tree one layer
+  }
+    
+  // find the sum of distances along the trunk path of the tree
+  double sumOfDistances = 0.0;
+  for (int i = 0; i < pathIndices.size() - 1; i++)
+    sumOfDistances += graph[i][i+1];
+
+  // check this to prevent a division by zero (in case all points are duplicates)
+  if (sumOfDistances == 0)
+  {
+    vtkErrorMacro("Minimum spanning tree path has distance zero. No parameters will be assigned. Check inputs!");
```

Comment:
"Check inputs!" - message to whom? what inputs? to what? refer to method name and specific arguments that may be wrong